### ISSUE TITLE
Some MiniZinc subsolvers give wrongly an unsatisfiable on a specific input file

### ISSUE THREAD

**Issue by rubenkindt** [CPMpy/cpmpy#160]
solver: Minizinc's subsolvers minizinc:float, minizinc:coinbc, minizinc:mip, minizinc:osicbc
CPMpy: version v0.9.9 commit e79b3afedc934a9437c2ddb3a9f54d7e2d7bd3ee
minizinc-python release version 0.7.0 commit {https://github.com/MiniZinc/minizinc-python/commit/a195cf63fcfbc98665d70ab64efb5424db25bd7e)

Some MiniZinc solvers did not complain about an "alldifferent()" in the problem but did give a UNSATISFIABLE with a runtime or 0 sec. Unfortunately, I could not remove any constraint from the original file even a shuffle of the constraints turns the problem into a feasible one.

[original.zip](https://github.com/CPMpy/cpmpy/files/10015191/original.zip)

```python 
from cpmpy import *

file = "original" # don't forget to unzip
m = Model().from_file(file)
m.solve(solver="minizinc:mip") # same with osicbc, float, coinbc
print(m.status())
```

expected behavior: 
That the mip solver complains about the 'alldifferent()' constraint in the problem or give any other status different from UNSATISFIABLE, since the correct solution is optimal.

bug found while working on my master thesis.
Edits: better title, added the shuffle sentence, changed "minimized" to "original" and added used minizinc-python version

**Comment by rubenkindt** [comment]
Looking at what 
```Python
from cpmpy.transformations.get_variables import get_variables, get_variables_model
for i in get_variables_model(m):
    print(i + ".lb="+str(i.lb) +",ub="+str(i.ub)+"name="+str(i.name))
```
produces, this bug most likely is the same one as issues #96 and #164. This could be fixed by pull #179

**Comment by Wout4** [comment]
this is indeed a bounds issue with minizinc, we check for that now and throw an error
